### PR TITLE
test/pybind/test_rados.py: tolerate TimedOut in test_ping_monitor

### PR DESCRIPTION
**Pull request by athanatos**
Fixes: http://tracker.ceph.com/issues/18529
Signed-off-by: Samuel Just <sjust@redhat.com>

### CODE REVIEW

**Comment by athanatos** [comment]
Tested: http://pulpito.ceph.com/samuelj-2017-01-14_22:21:18-rados-wip-sam-testing---basic-smithi/ failures seem unrelated

**Comment by tchaikov** [comment]
if `self.rados.ping_monitor()` times out, an exception would have been raised. but in this case, seems the pinged monitor returned an empty string, hence @None@ is returned. and it cannot be decoded by json.

i pasted the backtrace of the failed test at http://tracker.ceph.com/issues/18529#note-2

**Comment by jdurgin** [comment]
@tchaikov's analysis is correct - we're getting a 0 return value and empty output from ping_monitor(), not a TimedOut exception - I just reproduced this with ```ms inject socket failure = 30``` on the client side.

I think it's fine to work around this in the test, since ping_monitor is a pretty specialized command - just need to try again if we get an empty response rather than an exception, and same for the bash cephtool test

**Comment by athanatos** [comment]
Updated to fix the unit test at least

Full run: http://pulpito.ceph.com/samuelj-2017-01-20_21:05:28-rados-wip-sam-testing---basic-smithi/

@jdurgin 
Ready to merge.

File: src/test/pybind/test_rados.py [hunk 1]
```diff
@@ -144,7 +144,10 @@ def test_ping_monitor(self):
         ret, buf, out = self.rados.mon_command(json.dumps(cmd), b'')
         for mon in json.loads(buf.decode('utf8'))['mons']:
             while True:
-                buf = json.loads(self.rados.ping_monitor(mon['name']))
+                output = self.rados.ping_monitor(mon['name'])
```
Comment (tchaikov) [inline src/test/pybind/test_rados.py]: in `rados.pyx`, if the returning length is 0, `None` is returned. so probably, we should instead check like:
```python
if output:
   continue
buf = json.loads(output)
```

or be more specific:
```python
if output is None:
   continue
buf = json.loads(output)
```

Comment (athanatos) [inline src/test/pybind/test_rados.py]: Updateed

**Pull request merged**